src/renderer/screens/Operations/Operations.tsx

```diff
@@ -14,25 +14,44 @@ import { MultisigTransactionDS } from '@renderer/services/storage';
 import { useMultisigTx } from '@renderer/services/multisigTx/multisigTxService';
 import { Header } from '@renderer/components/common';
 import { useNetworkContext } from '@renderer/context/NetworkContext';
+import { MultisigEvent, MultisigTransaction } from '@renderer/domain/transaction';
+import { useMultisigEvent } from '@renderer/services/multisigEvent/multisigEventService';
 
 const Operations = () => {
   const { t, dateLocale } = useI18n();
   const { getActiveMultisigAccount } = useAccount();
   const { getLiveAccountMultisigTxs } = useMultisigTx();
   const { connections } = useNetworkContext();
+  const { getEventsByKeys } = useMultisigEvent();
 
   const account = getActiveMultisigAccount();
   const allTxs = getLiveAccountMultisigTxs(account?.accountId ? [account.accountId] : []);
 
   const [txs, setTxs] = useState<MultisigTransactionDS[]>([]);
+  const [events, setEvents] = useState<MultisigEvent[]>([]);
   const [filteredTxs, setFilteredTxs] = useState<MultisigTransactionDS[]>([]);
 
-  const groupedTxs = groupBy(filteredTxs, ({ dateCreated }) =>
-    format(new Date(dateCreated || 0), 'PP', { locale: dateLocale }),
-  );
+  const getEventByTransaction = (events: MultisigEvent[], tx: MultisigTransaction): MultisigEvent | undefined => {
+    return events.find(
+      (e) =>
+        e.txAccountId === tx.accountId &&
+        e.txChainId === tx.chainId &&
+        e.txCallHash === tx.callHash &&
+        e.txBlock === tx.blockCreated &&
+        e.txIndex === tx.indexCreated,
+    );
+  };
+
+  const groupedTxs = groupBy(filteredTxs, (tx) => {
+    const date = tx.dateCreated || getEventByTransaction(events, tx)?.dateCreated || Date.now();
+
+    return format(new Date(date), 'PP', { locale: dateLocale });
+  });
 
   useEffect(() => {
```

Comment:
To increase readability 
```suggestion
  useEffect(() => {
    const txsWithoutDate = allTxs.filter((tx) => !tx.dateCreated);
    getEventsByKeys(txsWithoutDate).then(setEvents);
    
    setTxs(allTxs.filter((tx) => connections[tx.chainId]));
    setFilteredTxs([]);
  }, [allTxs]);
```